src/R/Core/Impl/Tokens/Number.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Globalization;
+
+namespace Microsoft.R.Core.Tokens {
+    public static class Number {
+        private const NumberStyles _hexStyle = NumberStyles.HexNumber | NumberStyles.AllowHexSpecifier;
+        private const NumberStyles _integerStyle = NumberStyles.Integer;
+        private const NumberStyles _doubleStyle = NumberStyles.Number | NumberStyles.AllowExponent;
+
+        public static bool TryParse(string text, out double doubleResult, bool allowLSuffix = true) {
+            doubleResult = Double.NaN;
+            int sign = 1;
+
+            if (text.Length == 0) {
+                return false;
+            }
+            if (allowLSuffix && text[text.Length - 1] == 'L') {
+                text = text.Substring(0, text.Length - 1);
+            }
+
+            if (text.Length == 0) {
+                return false;
+            }
+            if (text[0] == '-' || text[0] == '+') {
```

Comment:
My 2c is that it should colorize in the same way that the actual R parser does things, so if it considers it an operator (which seems to be the case if you look at output of `parse(-1)`), then that should be it.

Also, in the future, when we have "Go to Definition", it will probably make sense to have it for operators as well, in which case treating `-` as a separate token would be good (it would go to `` base::`-` ``).

But sure, let's leave it for later.